### DISCUSSION TITLE
Fetch draft product

### DISCUSSION THREAD

**Discussion by vlefort** [gatsbyjs/gatsby#32092] [Ideas / Feature Requests]
Actually plugin only fetch active product.

It can be very useful to be able to fetch also draft products if wanted.

**Comment by sslotsky** [comment]
Thanks for the suggestion! We were already looking at pulling in draft content as part of supporting CMS Preview, but we can consider supporting opt-in draft products for regular builds as well. I'd love to know more about your use case first if you'd be willing to elaborate. For example, would you expect to see them mixed in with all the other products, or would you want to see them as a separate node type? Any other details you think would be pertinent?

**Comment by vlefort** [comment]
In production draft products should not be mixed to active products.
In developement or preview build see active and draft products is very interesting. If needed to have a specific page for draft products (for marketing or product creator) can also be interesting and it can be done easily by filtering them in GraphQL query (or even show a draft banner in product page for exemple).

**Comment by sahandnayebaziz** [comment]
It would be great to be able to toggle something like `includeDraftProducts: true` or something like that just to be able to get draft products as well in the mix. The problem I'm running into right now is not being able to support previewing draft products in a clients store.

**Comment by orinokai** [comment]
I also require this functionality. I have recently launched a Shopify store with a Gatsby frontend and now need to continue developing new features. I want to create a new draft product type that is not shown on the live store, but which I can test with on my local build. As @sahandnayebaziz mentioned, a parameter would be great, ideally one that I can set in gatsby-config.js using env vars.
